cmd/zed/agents/zfs_mod.c

```diff
@@ -83,6 +82,7 @@
 #include <sys/sunddi.h>
 #include <sys/sysevent/eventdefs.h>
 #include <sys/sysevent/dev.h>
+#include <thread_pool.h>
 #include <pthread.h>
```

Comment:
Tiny nit, you can probably now drop the `pthread.h` include since the code now no longer directly uses pthreads (only through the thread pool library)

Reply:
Dropping the `#include <pthread.h>` doesn't seem to result in any build error (it's included in `thread_pool.h` anyway), but since we are using `pthread_create()` to spawn `zfs_enum_pools()` it seems clearer to explicitly include the header file:

```
C symbol: pthread_create

  File                  Function            Line
0 zfs_agents.c          zfs_agent_init       311 if (pthread_create(&g_agents_tid, NULL, zfs_agent_consumer_thread,
1 zfs_mod.c             zfs_slm_init         861 if (pthread_create(&g_zfs_tid, NULL, zfs_enum_pools, NULL) != 0) {
2 zed_disk_event.c      zed_disk_event_init  374 if (pthread_create(&g_mon_tid, NULL, zed_udev_monitor, g_mon) != 0) {
3 thread_pool.c         create_worker        236 error = pthread_create(&thread, &tpool->tp_attr, tpool_worker, tpool);
```

Reply:
I'm happy to leave it then.